presto-spi/src/main/java/com/facebook/presto/spi/block/RunLengthEncodedBlock.java

```diff
@@ -139,42 +143,49 @@ public Block copyRegion(int positionOffset, int length)
     @Override
     public int getSliceLength(int position)
     {
+        checkReadablePosition(position);
```

Comment:
Those checks seem to be missing before your PR. Good that you added them, but the commit message reads `Changes for representing 'all null values' in a RunLenghtEncodedBlock`. Can you add those checks in a separate preparatory (first) commit titled `Add position checks in RunLenghtEncodedBlock`?

(btw. my IntelliJ's annotate shows your commits as authored by `D`. Could it be more informative?)

Reply:
@findepi should it be a separate PR or ?

Reply:
I meant separate commit, might be same PR

Reply:
Okay.